### ISSUE TITLE
make constevaler use host native flags

### ISSUE THREAD

**Issue by powderluv** [iree-org/iree#14200]
### Request description

As models get larger look into adding host specific flags for consteval etc. 

Related discussion https://discord.com/channels/689900678990135345/760577505840463893/1121656241190621194

**Comment by allieculp** [comment]
@stellaraccident Bumping this up, please take a look!

**Comment by stellaraccident** [comment]
Yes, needs to be done but both a non trivial sequence of work and not P1 right now

**Comment by ScottTodd** [comment]
Was this fixed by https://github.com/openxla/iree/pull/14530 / https://github.com/openxla/iree/pull/14516 ? Or is there more needed in-tree to support this?
